xml/dhcp_vendor_class.xml

```diff
@@ -578,6 +578,15 @@
     <param pos="0" name="os.certainty" value="0.8"/>
   </fingerprint>
 
+  <fingerprint pattern="^Avaya E129 dslforum.org$">
+    <description>Avaya E129 IP Phone</description>
+    <example>Avaya E129 dslforum.org</example>
+    <param pos="0" name="hw.vendor" value="Avaya"/>
+    <param pos="0" name="hw.product" value="E129"/>
+    <param pos="0" name="hw.device" value="VoIP"/>
+    <param pos="0" name="os.vendor" value="Avaya"/>
```

Comment:
From a quick review of `h:avaya:*` and `o:avaya:*` the CPE strings I see reflect a relationship between the hardware products and operating system firmware. These are in the format `o:avaya:{hw.product}_firmware` so I would add the following parameter:
```
<param pos="0" name="os.product" value="{hw.product} Firmware"/>
```

Reply:
K, sounds good.